docs/examples/quickstart/test_quickstarts.rb

```diff
@@ -0,0 +1,143 @@
+# frozen_string_literal: true
+
+require 'English'
+require 'fileutils'
+require 'timeout'
+
+require 'bundler'
+
+CURL_ERROR = "curl: (7) Failed to connect to localhost port 5050: Connection refused\n"
+CURL_EMPTY = "curl: (52) Empty reply from server\n"
+
+quickstarts = [
+  { lang: 'java', setup: 'make build', server: 'make run' },
+  { lang: 'nodejs', setup: 'npm i', server: 'npm start' },
+  { lang: 'python', setup: 'pip install --upgrade -r requirements.txt', server: 'python server.py' },
+  { lang: 'ruby', setup: 'bundle', server: 'bundle exec ruby server.rb' },
+  { lang: 'rust', setup: 'cargo build', server: 'cargo run' }
+]
+
+def start_server(server, user, expense_id)
+  server = spawn server
+  received = CURL_ERROR
+  while [CURL_ERROR, CURL_EMPTY].include? received
+    sleep 0.5
+    received = `curl -sSH "user: #{user}" localhost:5050/expenses/#{expense_id} 2>&1`
+  end
+  [server, received]
+end
+
+def ensure_port_5050_is_open
+  until (server = `lsof -ti :5050 2>&1`.split.first.to_i).zero?
+    sleep 0.5
+    Process.kill 'TERM', server
+  end
+rescue Errno::ESRCH => e
+  puts "#{e}: #{server}"
+end
+
+def kill_server(server)
+  return if server.nil?
```

Comment:
Should this be an assert? When would `server` be nil?

Reply:
It bit me once, so I added the check. Not sure if it's still necessary (reading the code it doesn't seem like it should be possible), but I at least moved the check to a more appropriate place in 5ff8521.